### ISSUE TITLE
False positive with useless-super-delegation for keyword args with dict defaults

### ISSUE THREAD

**Issue by filip-zyzniewski** [PyCQA/pylint#3773]
Problem similar to #2022 .

### Steps to reproduce

(I know, mutable default is a bad idea)

```python
# uselesssuper.py
class MyBase:
    def method(self, *, arg={}):
        self.x = arg

class MyChild(MyBase):
    def method(self, *, arg={'foo': 'bar'}):
        super().method(arg=arg)

myobj = MyChild()
myobj.method()
print(myobj.x)
```

### Current behavior

```shell
$ python3 uselesssuperdict.py 
{'foo': 'bar'}
$ pylint uselesssuperdict.py --disable=all --enable=useless-super-delegation
************* Module uselesssuperdict
uselesssuperdict.py:7:4: W0235: Useless super delegation in method 'method' (useless-super-delegation)

------------------------------------------------------------------
Your code has been rated at 8.89/10 (previous run: 8.89/10, +0.00)

$ 
```

### Expected behavior

No issue

### pylint --version output

```shell
$ pylint --version
pylint 2.5.3
astroid 2.4.2
Python 3.8.5 (default, Jul 21 2020, 10:42:08) 
[Clang 11.0.0 (clang-1100.0.33.17)]
$
```

**Comment by hippo91** [comment]
@filip-zyzniewski thanks for the report. I made PR #3778 that should correct this.
